addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -1934,14 +1934,15 @@ var SnippetsMenu = Widget.extend({
      *        If no element is given, all the editors are destroyed.
      */
     _destroyEditors: async function ($el) {
-        const proms = _.map(this.snippetEditors, async function (snippetEditor) {
-            if ($el && !$el.has(snippetEditor.$target).length) {
-                return;
-            }
-            await snippetEditor.cleanForSave();
-            snippetEditor.destroy();
+        const aliveEditors = this.snippetEditors.filter((snippetEditor) => {
+            return !$el || $el.has(snippetEditor.$target).length;
         });
-        await Promise.all(proms);
+        const cleanForSavePromises = aliveEditors.map((snippetEditor) => snippetEditor.cleanForSave());
+        await Promise.all(cleanForSavePromises);
+
+        for (const snippetEditor of aliveEditors) {
+            snippetEditor.destroy();
+        }
```

Comment:
Seems good, should not it be done in stable though ?

Reply:
You mean you would backport the fix ?

Reply:
PR in 15.0: https://github.com/odoo/odoo/pull/83120

Reply:
Arguably, I would consider to do it even before that (even when the `cleanForSave` was not `async` yet, calling methods of destroyed objects is just a cause for problems). But I guess 15.0 is the minimum IMO, we allow `cleanForSave` to be `async` to be able to make RPC... if that can make the editor be stuck, it's just a real bug :+1:

Note: also I would just have retarget this PR instead of spreading conversations across multiple ones, repinging teams, etc :wink: